source/s3_meta_request.c

```diff
@@ -663,6 +677,10 @@ void aws_s3_meta_request_sign_request_default(
         return;
     }
 
+    if (meta_request->checksum_algorithm != AWS_SCA_NONE && s_is_put_request(meta_request) &&
```

Comment:
nit: one condition per line would be more readable.
also comment on why it needs to be done would be helpful

Reply:
formatter doesn't like separating conditions here.